pkg/package-server/storage/packagemanifest/watcher.go

```diff
@@ -0,0 +1,102 @@
+package packagemanifest
+
+import (
+	"context"
+	"sync"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/watch"
+
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/package-server/apis/packagemanifest/v1alpha1"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/package-server/provider"
+)
+
+type Watcher struct {
```

Comment:
Based off of the [example in k8s apimachinery](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/vendor/k8s.io/apimachinery/pkg/watch/watch.go#L159).